test/integration/util/util.go

```diff
@@ -192,7 +196,13 @@ func PodDeleted(c clientset.Interface, podNamespace, podName string) wait.Condit
 // SyncInformerFactory starts informer and waits for caches to be synced
 func SyncInformerFactory(testCtx *TestContext) {
 	testCtx.InformerFactory.Start(testCtx.Ctx.Done())
+	if testCtx.DynInformerFactory != nil {
+		testCtx.DynInformerFactory.Start(testCtx.Ctx.Done())
+	}
 	testCtx.InformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
+	if testCtx.DynInformerFactory != nil {
+		testCtx.DynInformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
+	}
```

Comment:
this can be:

```suggestion
	testCtx.InformerFactory.Start(testCtx.Ctx.Done())
	testCtx.InformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
	if testCtx.DynInformerFactory != nil {
		testCtx.DynInformerFactory.Start(testCtx.Ctx.Done())
		testCtx.DynInformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
	}
```

Reply:
This was intentional :) discussed in https://github.com/kubernetes/kubernetes/pull/105016#discussion_r709603422

Reply:
I haven't played with that, so might be significant for bigger caches.